lib/unison-sqlite/src/Unison/Sqlite/Sql2.hs

```diff
@@ -0,0 +1,263 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Unison.Sqlite.Sql2
+  ( Sql2 (..),
+    sql2,
+
+    -- * Exported for testing
+    internalParseSql,
+  )
+where
+
+import Control.Lens ((%=), (<>=))
+import qualified Control.Monad.Trans.State.Strict as State
+import qualified Data.Char as Char
+import Data.Generics.Labels ()
+import qualified Data.Text as Text
+import qualified Database.SQLite.Simple as Sqlite.Simple
+import qualified Language.Haskell.TH as TH
+import qualified Language.Haskell.TH.Quote as TH
+import qualified Text.Builder
+import qualified Text.Builder as Text (Builder)
+import qualified Text.Megaparsec as Megaparsec
+import qualified Text.Megaparsec.Char as Megaparsec
+import Unison.Prelude
+
+-- | A SQL query.
+data Sql2 = Sql2
+  { query :: Text,
+    params :: [Sqlite.Simple.NamedParam]
```

Comment:
I like the simplicity of using sqlite's named parameters. 

A couple of drawbacks that come to mind are: 

1. If we want to add splicing someday then a disadvantage compared to accumulating a `[SqlData]` is that you must be aware of names taken by other splices
2. We pay the small runtime cost of looking up the param index although we know it at compile time

Reply:
Oh, (1) is a good point, and that will make debug logs better as well. I'll think I'll address that in a separate PR, though. The use of named params here is just for convenience, but it's not that inconvenient to use `[SqlData]` instead.